tensorflow/python/summary/summary_iterator.py

```diff
@@ -301,15 +306,16 @@ def __init__(self, queue, ev_writer, flush_secs):
     """
     threading.Thread.__init__(self)
     self.daemon = True
+    self._stop_flag = False
     self._queue = queue
     self._ev_writer = ev_writer
     self._flush_secs = flush_secs
     # The first event will be flushed immediately.
     self._next_event_flush_time = 0
 
-  def run(self):
+  def _run(self):
     while True:
-      event = self._queue.get()
+      event = self._queue.get(False)
```

Comment:
Calling `self._queue.get(False)` when the queue is empty will result in this thread busy-waiting for a new element (or for the queue to be closed), which will have unacceptable performance.

Instead of relying on exceptions, can you implement the sentinel object version?